gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemNewIntegrationTest.java

```diff
@@ -711,6 +711,42 @@ private void delete_directory(boolean parallelStatus) throws Exception {
     assertThat(gcsFs.exists(bucketUri.resolve(dirObject + "/d1"))).isFalse();
   }
 
+  @Test
+  public void delete_inferred_directory() throws Exception {
+    GoogleCloudStorageFileSystemOptions gcsFsOptions =
+        newGcsFsOptions().setStatusParallelEnabled(false).build();
+
+    TrackingHttpRequestInitializer gcsRequestsTracker =
+        new TrackingHttpRequestInitializer(httpRequestsInitializer);
+    GoogleCloudStorageFileSystem gcsFs = newGcsFs(gcsFsOptions, gcsRequestsTracker);
+
+    String bucketName = gcsfsIHelper.sharedBucketName1;
+    URI bucketUri = new URI("gs://" + bucketName + "/");
+    String dirObject = getTestResource();
+
+    gcsfsIHelper.createObjects(bucketName, dirObject + "/d1/d2");
+
+    gcsFs.delete(bucketUri.resolve(dirObject + "/d1/"), /* recursive= */ true);
+
+    assertThat(gcsRequestsTracker.getAllRequestStrings())
+        .containsExactly(
+            listRequestWithTrailingDelimiter(
+                bucketName, dirObject + "/d1/", /* maxResults= */ 1, /* pageToken= */ null),
+            listRequestString(
+                bucketName,
+                /* flatList= */ true,
+                /* includeTrailingDelimiter= */ null,
+                dirObject + "/d1/",
+                "bucket,name,generation",
+                /* maxResults= */ 1024,
+                /* pageToken= */ null),
+            deleteRequestString(bucketName, dirObject + "/d1/d2", /* generationId= */ 1),
+            getRequestString(bucketName, dirObject + "/"),
+            uploadRequestString(bucketName, dirObject + "/", /* generationId= */ null));
```

Comment:
May you pre-create `dirObject + "/"` so there will be no implicit directory repair?

Reply:
Done